### ISSUE TITLE
The RJPP_BRANCH env variable is set to the source's branch, not the remote file's branch

### ISSUE THREAD

**Issue by Angelin01** [jenkinsci/remote-file-plugin#37]
It appears the variable that was supposed to indicate the remote file branch (or, hopefully, also tags, which is our use case), actually points to the source branch.

[This line in particular is suspicious](https://github.com/jenkinsci/remote-file-plugin/blob/4c06b541043252061ae113ba82c0936ca733d523/src/main/java/org/jenkinsci/plugins/workflow/multibranch/extended/scm/ExtendedSCMBinder.java#L49)

Steps to reproduce the behaviour:
1. Create a job using the Remote Jenkinsfile build configuration
2. Change the default branch to any other branch
3. Use the `RJPP_BRANCH` variable in the pipeline
4. Notice that the value of `$RJPP_BRANCH` is always the source branch (the same as `BRANCH_NAME`).

This is the pipeline on my master branch:
```
pipeline {
    agent any

    stages {
        stage('Test') {
            steps {
                echo 'Hello!'
            }
        }
        stage('Branch') {
            steps {
                echo "Branch is: ${env.RJPP_BRANCH}"
            }
        }
    }
}
```

This is the pipeline on my "not-master" branch:

```
pipeline {
    agent any

    stages {
        stage('Test') {
            steps {
                echo 'Hello! This is the not-master pipeline!'
                echo "Current source branch is ${env.BRANCH_NAME}"
            }
        }
        stage('Branch') {
            steps {
                echo "Current RJPP_BRANCH branch is: ${env.RJPP_BRANCH}"
            }
        }
    }
}
```

Screenshots:
![image](https://user-images.githubusercontent.com/17818024/128772374-a47c627c-47eb-4269-b042-2226a2723412.png)

![image](https://user-images.githubusercontent.com/17818024/128772399-0e9402f4-49c1-487d-aa31-0b2d952cfcd0.png)

![image](https://user-images.githubusercontent.com/17818024/128772495-b3f2eccc-56f7-4a8c-ac47-221455eeb0af.png)

**Comment by Angelin01** [comment]
As a side note, it looks like there is very little documentation regarding the `RJPP_BRANCH` variable. Maybe some should be added, both here and in the Jenkins plugin page.

**Comment by matpag** [comment]
I thinks this is the same as #36

**Comment by Angelin01** [comment]
> 
> 
> I thinks this is somewhat related to #36

@matpag I don't actually experience YOUR (#36) bug. Using both "Gitlab Project" and the simple generic "git" I can get git tags properly using `refs/tags/v.1.1`, but I've never used the "Fallback branch" field, only the Discover Tags option + direct ref.

**Comment by matpag** [comment]
Where did you specify the tags to download the remote Jenkinsfile?

**Comment by aytuncbeken** [comment]
Hi @Angelin01 
Yes this is a bug, this parameters ```RJPP_BRANCH``` should always show the branch of the Jenkinsfile which is checked out. 
I will make this fix.
Thanks for reporting!

**Comment by aytuncbeken** [comment]
Hi @Angelin01 
Fixed with 1.21 version
